src/flownet/utils/plot_results.py

```diff
@@ -0,0 +1,372 @@
+    """
+
+    parser = argparse.ArgumentParser(
+        prog=("Simple tool to plot FlowNet ensembles simulation results.")
+    )
+    parser.add_argument(
+        "vectors",
+        type=str,
+        nargs="+",
+        help="One or more vectors to plot separated by spaces. Example: WOPR:WELL1 FOPR",
+    )
+    parser.add_argument(
+        "-prior",
+        type=str,
+        nargs="+",
+        help="One or more paths to prior ensembles separated by a space."
+        "The path should include a '%d' which indicates the realization number.",
+    )
+    parser.add_argument(
+        "-posterior",
+        type=str,
+        nargs="+",
+        help="One or more paths to posterior ensembles separated by a space."
+        "The path should include a '%d' which indicates the realization number.",
+    )
+    parser.add_argument(
+        "-reference_simulation",
+        "-r",
+        type=pathlib.Path,
+        help="Path to the reference simulation case.",
+    )
+    parser.add_argument(
+        "-ymin",
+        type=float,
+        default=[0],
+        nargs="+",
+        help="One or #vectors minimum y values.",
+    )
+    parser.add_argument(
+        "-ymax",
+        type=float,
+        default=[1000],
+        nargs="+",
+        help="One or #vectors maximum y values.",
+    )
+    parser.add_argument(
+        "-units",
+        type=str,
+        default=[""],
+        nargs="+",
+        help="One or #vectors unit labels.",
+    )
+    parser.add_argument(
+        "-prior_alphas",
+        type=float,
+        default=[0.1],
+        nargs="+",
+        help="One or #prior ensembles alpha (transparency) values.",
+    )
+    parser.add_argument(
+        "-posterior_alphas",
+        type=float,
+        default=[0.1],
+        nargs="+",
+        help="One or #posterior ensembles alpha (transparency) values.",
+    )
+    parser.add_argument(
+        "-prior_colors",
+        type=str,
+        default=["gray"],
+        nargs="+",
+        help="One or #prior ensembles colors.",
+    )
+    parser.add_argument(
+        "-posterior_colors",
+        type=str,
+        default=["blue"],
+        nargs="+",
+        help="One or #posterior ensembles colors.",
+    )
+    parser.add_argument(
+        "-reference_simulation_color",
+        type=str,
+        default="red",
+        help="The reference simulation color.",
+    )
+    parser.add_argument(
+        "-vertical_lines",
+        type=lambda s: datetime.strptime(s, "%Y-%m-%d"),
+        default=None,
+        nargs="+",
+        help="One or more date (YYYY-MM-DD) to add vertical lines in the plot.",
+    )
+    parser.add_argument(
+        "-ertobs",
+        type=pathlib.Path,
+        default=None,
+        help="Path to an ERT observation file.",
+    )
+    args = parser.parse_args()
+
+    check_args(args)
+
+    prior_data = build_ensemble_df_list(args.prior, args.vectors)
+    posterior_data = build_ensemble_df_list(args.posterior, args.vectors)
+
```

Comment:
Same here. You can end up trying to iterate over `None` inside `build_ensemble_df_list`.
```suggestion
prior_data = build_ensemble_df_list(args.prior, args.vectors) if args.prior else None
posterior_data = build_ensemble_df_list(args.posterior, args.vectors) if args.posterior else None
```

Reply:
Fixed by checking in build_ensemble_df_list if the input is None, if so I return an empty list (to make sure I at least return a list which is expected later on).